File: src/main/java/com/qtech/bubbles/registry/RegistryEntry.java
package com.qtech.bubbles.registry;

import com.qtech.bubbles.common.IRegistryEntry;
import com.qtech.bubbles.common.ResourceEntry;

import java.util.Objects;

public abstract class RegistryEntry implements IRegistryEntry {
    private ResourceEntry registryName = null;

    public void setRegistryName(String namespace, String name) {
        registryName = new ResourceEntry(namespace, name);
    }

    public boolean isTempRegistryName() {
        return registryName.namespace() == null;
    }

    public void updateRegistryName(String namespace) {
        if (registryName.namespace() == null) {
            registryName = registryName.withNamespace(namespace);
        }
    }

    public ResourceEntry getRegistryName() {
        return registryName;
    }

    public void setRegistryName(String name) {
        registryName = new ResourceEntry(null, name);
    }

    public void setRegistryName(ResourceEntry name) {
        this.registryName = name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        RegistryEntry that = (RegistryEntry) o;
        return Objects.equals(registryName, that.registryName);
    }

    @Override
    public int hashCode() {
        return Objects.hash(registryName);
    }
}

File: src/main/kotlin/com/qtech/bubbles/common/RegistryEntry.kt
package com.qtech.bubbles.common

import java.util.*

abstract class RegistryEntry : IRegistryEntry {
    final override var registryName: ResourceLocation? = null

    fun setRegistryName(namespace: String?, name: String) {
        registryName = ResourceLocation(namespace, name)
    }

    override val isTempRegistryName: Boolean
        get() = registryName?.namespace == null

    override fun updateRegistryName(namespace: String) {
        if (registryName?.namespace == null) {
            registryName = registryName!!.withNamespace(namespace)
        }
    }

    fun setRegistryName(name: String) {
        registryName = ResourceLocation(null, name)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as RegistryEntry
        return registryName == that.registryName
    }

    override fun hashCode(): Int {
        return Objects.hash(registryName)
    }
}